fix(ivy): ngtsc fails building flat module out on windows

devversion opened the pull request:
`ngtsc` currently fails building a flat module out file on Windows because it generates an invalid flat module TypeScript source file. e.g:

```ts
5 export * from './C:\Users\Paul\Desktop\test\src\export';
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

This is because `path.posix.relative` does not properly with non-posix paths, and only expects posix paths in order to work.

mary-poppins commented:
You can preview b011336 at https://pr27993-b011336.ngbuilds.io/.

mary-poppins commented:
You can preview b4ee908 at https://pr27993-b4ee908.ngbuilds.io/.

mary-poppins commented:
You can preview fd3f12c at https://pr27993-fd3f12c.ngbuilds.io/.

mary-poppins commented:
You can preview 06c8106 at https://pr27993-06c8106.ngbuilds.io/.

mary-poppins commented:
You can preview c0602cc at https://pr27993-c0602cc.ngbuilds.io/.

mary-poppins commented:
You can preview 6863c3f at https://pr27993-6863c3f.ngbuilds.io/.

alxhub commented on packages/compiler-cli/src/ngtsc/entry_point/src/generator.ts:
```diff
@@ -27,9 +27,13 @@ export class FlatIndexGenerator implements ShimGenerator {
-    const relativeEntryPoint = './' +
-        path.posix.relative(path.posix.dirname(this.flatIndexPath), this.entryPoint)
-            .replace(/\.tsx?$/, '');
+    // We need to normalize the path separators manually instead of using path.posix, because
```
I would prefer that we normalize these values at the source, rather than when we go to use them.

devversion replied:
Done.

mary-poppins commented:
You can preview 21eefa6 at https://pr27993-21eefa6.ngbuilds.io/.

mary-poppins commented:
You can preview 8f0c03e at https://pr27993-8f0c03e.ngbuilds.io/.

mary-poppins commented:
You can preview b644f71 at https://pr27993-b644f71.ngbuilds.io/.

alfaproject commented on packages/compiler-cli/src/ngtsc/entry_point/src/generator.ts:
```diff
@@ -45,3 +44,11 @@ export * from '${relativeEntryPoint}';
+ * Normalizes the specified path by replacing backslash separators with Posix
+ * forward slash separators.
+ */
+function normalizeSeparators(path: string): string {
```
Left over?

devversion replied:
Yeah it is. I've been still working on that PR, but thanks!

<sup>Windows user; so the Bazel ngtsc tests don't run locally yet</sup>

devversion committed:
fix(ivy): ngtsc fails building flat module out on windows

`ngtsc` currently fails building a flat module out file on Windows because it generates an invalid flat module TypeScript source file. e.g:

```ts
5 export * from './C:\Users\Paul\Desktop\test\src\export';
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

This is because `path.posix.relative` does not properly with non-posix paths, and only expects posix paths in order to work.

mary-poppins commented:
You can preview 796dcaa at https://pr27993-796dcaa.ngbuilds.io/.

mary-poppins commented:
You can preview ed97cd2 at https://pr27993-ed97cd2.ngbuilds.io/.

devversion committed:
fixup! fix(ivy): ngtsc fails building flat module out on windows

Fix linting

alxhub commented:
[Presubmit](http://test/OCL:229801886:BASE:229802006:1547758690237:70eb5c50)

alxhub commented:
And [rerunning the presubmit](https://test.corp.google.com/ui#id=OCL:229801886:BASE:229948796:1547835186444:10ec8133) due to a large number of preexisting broken tests (not the fault of this PR).

alxhub commented:
And of course over the long weekend that presubmit expired from the results cache, so [here's yet another one](http://test/OCL:229801886:BASE:230358029:1548179482202:f89e4718)!
